src/components/form/HdRange.vue

```diff
@@ -0,0 +1,329 @@
+<template>
+  <div :class="fieldClasses" class="range field">
+    <input
+      type="range"
+      v-model.number="currentValue"
+      :id="name"
+      :name="name"
+      :required="required"
+      :disabled="disabled"
+      :min="minValue"
+      :max="maxValue"
+      :step="rangeStep"
+      @focus="focusHandler"
+      @blur="blurHandler"
+    />
+    <div class="range__decoration" ref="decoration">
+      <div class="range__progress" ref="progress" />
+    </div>
+    <ul v-if="displayStepBullets" class="range__steps">
+      <li v-for="(steps, i) in stepsAmount" :key="i" />
+    </ul>
+    <div class="range__thumb" ref="thumb">
+      <div class="range__thumb__inner" ref="thumbInner">
+        <div class="range__thumb__bullet" />
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'HdRange',
+  props: {
+    name: {
+      type: String,
+      default: 'range',
+    },
+    required: {
+      type: Boolean,
+      default: false,
+    },
+    disabled: {
+      type: Boolean,
+      default: false,
+    },
+    minValue: {
+      type: Number,
+      default: 0,
+    },
+    maxValue: {
+      type: Number,
+      default: 100,
+    },
+    rangeStep: {
+      type: Number,
+      default: 1,
+    },
+    value: {
+      type: Number,
+      default: 0,
+    },
+    displayStepBullets: {
+      type: Boolean,
+      default: true,
+    },
```

Comment:
I would add a test to cover this property as it is not tested anywhere and apparently, as pointed out by @SinisaG, there is a bug for this guy.

Reply:
I do agree.